mmdet/apis/train.py

```diff
@@ -68,6 +68,51 @@ def set_random_seed(seed, deterministic=False):
         torch.backends.cudnn.benchmark = False
 
 
+def scale_lr(cfg, logger):
+    """Automatically scaling LR according to GPU number and sample per GPU.
+
+    Args:
+        cfg (config): training config.
+        logger (logging.Logger): logger.
+    """
+    enable_auto_scale_lr = cfg.get('enable_auto_scale_lr', False)
+    if enable_auto_scale_lr is False:
+        logger.info('You disabled automatically scaling LR.')
+        return
+
+    # default batch size of pre-train model
+    default_batch_size = cfg.get('default_batch_size', None)
+    assert default_batch_size is not None
+
+    # calculate the batch size
+    num_gpus = len(cfg.gpu_ids)
+    batch_size = num_gpus * cfg.data.samples_per_gpu
+
+    logger.info(f'You are using {num_gpus} GPU(s) '
+                f'and {cfg.data.samples_per_gpu} samples per GPU. '
+                f'Total batch size is {batch_size}.')
+
+    if batch_size != default_batch_size:
+        # Get original LR from config
+        original_lr = cfg.optimizer.get('lr', 0)
+        assert original_lr != 0
+
+        # scale LR with
+        # [linear scaling rule](https://arxiv.org/abs/1706.02677)
+        scaled_lr = (batch_size / default_batch_size) * original_lr
+        cfg.optimizer.update({'lr': scaled_lr})
+
+        logger.info('Since default batch size of this model '
+                    f'is {default_batch_size}, '
+                    'automatically scaling LR '
```

Comment:
LR has been automatically scaled from {} to {}

Reply:
Done

Reply:
Done